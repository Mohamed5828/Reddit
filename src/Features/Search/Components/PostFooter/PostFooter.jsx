import { Footer } from "./PostFooter.styled";

/**
 * Component that contains the Post Header Footer.
 *
 * @Component
 * @param {object} postfooter - the post footer object
 * @returns {React.Component}
 */

const PostFooter = ({ postfooter }) => {
  if (postfooter) {
    return (
      <Footer>
        <span className="upvote">
          {postfooter.upVotes}
          <span className="upvote-units">k</span> upvotes
        </span>
        <span className="comment">
          {postfooter.Comments}
          <span className="comment-units">k</span> comments
        </span>
      </Footer>
    );
  }
};

export default PostFooter;
