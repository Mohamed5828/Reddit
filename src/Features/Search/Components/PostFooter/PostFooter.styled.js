import styled from "styled-components";
export const Footer = styled.div`
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 16px;
  font-size: 12px;
  font-weight: 400;
  line-height: 16px;
  color: ${({ theme }) => theme.color.muted};
  display: flex;
  span.comment,
  span.upvote {
    display: block;
    margin-right: 12px;
  }
`;
