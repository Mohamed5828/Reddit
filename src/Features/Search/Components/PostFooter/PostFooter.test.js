import { shallow } from "enzyme";
import PostFooter from "./PostFooter";
describe("Post Footer", () => {
  it("this is a test for Post Footer", () => {
    expect(shallow(<PostFooter />)).toMatchSnapshot();
  });
});
