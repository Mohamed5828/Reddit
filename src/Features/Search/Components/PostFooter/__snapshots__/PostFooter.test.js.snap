// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Footer this is a test for Post Footer 1`] = `
<styled.div>
  <span>
    1k upvotes
  </span>
  <span>
    1.7k comments
  </span>
</styled.div>
`;
